doc/example/gradient_check.ipynb

```diff
@@ -0,0 +1,894 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "# Gradient checks\n",
+    "\n",
+    "It is best practice to do gradient checks before and after gradient-based optimization.\n",
+    "Here we show, how to use the gradient check methods that are implemented in pyPESTO, using the finite differences (FD) method as a comparison.\n",
+    "There is a trade-off between the quality of the approximation and numerical noise, so it is recommended to try different FD step sizes.  \n",
+    "Most importantly, test your gradients using the settings you will use later on."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "import benchmark_models_petab as models\n",
+    "import pypesto.optimize as optimize\n",
+    "import pypesto.petab\n",
+    "\n",
+    "np.random.seed(2)\n",
+    "\n",
+    "import seaborn as sns"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "#### Set up an example problem\n",
```

Comment:
I think the heading levels aren't consistent yet (https://pypesto--1497.org.readthedocs.build/en/1497/example/gradient_check.html)